core/src/main/scala/org/apache/spark/sql/delta/Checkpoints.scala

```diff
@@ -290,18 +290,39 @@ object Checkpoints extends DeltaLogging {
     // log store and decide whether to use rename.
     val useRename = deltaLog.store.isPartialWriteVisible(deltaLog.logPath, hadoopConf)
 
-    val checkpointRowCount = spark.sparkContext.longAccumulator("checkpointRowCount")
-    val numOfFiles = spark.sparkContext.longAccumulator("numOfFiles")
+    val sessionConf = spark.sessionState.conf
+    val maxActionsPerFile = sessionConf.getConf(DeltaSQLConf.DELTA_CHECKPOINT_MAX_ACTIONS_PER_FILE)
+
+    val res = snapshot.stateDF
+      .groupBy()
+      .agg(count("*").alias("rowCount"), count("add").alias("numOfFiles"))
+      .collect()
+      .head
```

Comment:
This will trigger an extra Spark job to calculate the `rowCount`. Can we use an approximate value such as
```
val checkpointPartSize = ...
val estimatedNumParts = math.ceil(
        (snapshot.numOfFiles + snapshot.numOfRemoves).toDouble / checkpointPartSize
      ).toLong
```
to avoid launching a new Spark job?

Reply:
Ah yeah good call

Reply:
Updated to use just those values. Thought about always adding 2 for protocol + metadata to be more accurate but figured it didn't really matter since the files should dominate the overall count